modules/labs-react/common/spec/useThemeRTL.spec.ts

```diff
@@ -138,8 +138,7 @@ describe('styles', () => {
 
         const {result} = renderHook(() => useThemeRTL());
         const cssProp = result.current.themeRTL(style);
-
-        expect((cssProp['&.hover'] as CSSObject).paddingLeft).toEqual(expectedPadding);
+        expect((cssProp['&:hover'] as CSSObject).paddingLeft).toEqual(expectedPadding);
```

Comment:
this is not correct, but it would fail the test.

Reply:
The error I see in the logs is:
```sh
TypeError: Cannot read property 'paddingRight' of undefined
```
That doesn't mean `&:hover` isn't there, just that TypeScript is saying `&:hover` isn't a guaranteed key from the object that `themeRTL` returns. When we look at `themeRTL`, all that's guaranteed from function is an empty object ([see here](https://github.com/Workday/canvas-kit/blob/master/modules/labs-react/common/lib/theming/useThemeRTL.ts#L64)). We know the function will pass along `&:hover`, but TS doesn't.

So you need to tell TypeScript that the result of passing `style` to `themeRTL` will return an object containing the key `&:hover`. You can do that one of two ways by modifying line 140.


Option 1:
```ts
const cssProp = result.current.themeRTL(style) as { ['&hover:']: { paddingLeft: string } };
```

Option 2:
```ts
const cssProp = result.current.themeRTL(style) as typeof style;
```

Either is fine. I verified both work locally. It's up to you how you want to solve it.

Reply:
You'll also need to fix the next RTL test `should flip and convert hover styles`. But you'll need to cast the type on line 165 as this:

```ts
const cssProp = result.current.themeRTL(style) as { [':&hover']: { paddingRight: string }};
```

Same issue, but this time it will flip the style from `paddingLeft` to `paddingRight`, and we need to tell `TS` that `[:hover]` will be there and contain the key `paddingRight`.

Reply:
You'll also need to fix the next RTL test `should flip and convert hover styles`. But you'll need to cast the type on line 165 as this:

```ts
const cssProp = result.current.themeRTL(style) as { [':&hover']: { paddingRight: string }};
```

Same issue, but this time it will flip the style from `paddingLeft` to `paddingRight`, and we need to tell TS that `[:hover]` will be there and contain the key `paddingRight`.

Reply:
Another problem with the test on line 167 is the key is `['&.hover']` and not `['&:hover']`.

It should read:

```ts
expect((cssProp['&:hover'] as CSSObject).paddingRight).toEqual(expectedPadding);
```

Reply:
thank you!!

Reply:
I'm not sure why `useThemeRTL` doesn't return a function with a more precisely typed output.

```tsx
const themeRTL: <S extends CSSProperties>(styles: S) => S = useMemo(() => {
```

Then:
<img width="292" alt="image" src="https://user-images.githubusercontent.com/338257/194947446-8fcea15d-bb4a-42cb-85de-38feb112683c.png">

Reply:
Take out this line in this spec file:
```tsx
_styleRewriteFn: convertToStaticStates,
```

This is converting `:hover` to `.hover` which we do not need for these tests.

Reply:
Also, if these tests fail, you can get less than helpful error messages:
```
TypeError: Cannot read properties of undefined (reading 'paddingRight')
```

It is better to use more specific matchers to maximize failure messages. For example:
```tsx
expect(cssProp['&:hover']).toHaveProperty('paddingLeft', expectedPadding);
```

The failure looks like this:
<img width="550" alt="image" src="https://user-images.githubusercontent.com/338257/194949798-0e13523e-7e52-40c2-85f2-166ef0601a7d.png">

Reply:
thank you for that!